src/rust/engine/task_executor/src/lib.rs

```diff
@@ -139,8 +135,8 @@ impl Executor {
       workunit_store::set_thread_parent_id(workunit_parent_id);
       f()
     })
-    .map(|res| res.unwrap())
-    .boxed()
+    .await
+    .unwrap()
```

Comment:
Does spawn blocking always return Ok(R)?

Reply:
See the comment a few lines up: panics on background threads propagate this way... if a background thread panics, we want to panic too.